parser.go

```diff
@@ -200,18 +227,37 @@ func (p *Parser) ParseUnverified(tokenString string, claims Claims) (token *Toke
 // take into account whether the [Parser] is configured with additional options,
 // such as [WithStrictDecoding] or [WithPaddingAllowed].
 func (p *Parser) DecodeSegment(seg string) ([]byte, error) {
-	encoding := base64.RawURLEncoding
+	var encoding Base64Encoding
+	if p.rawUrlBase64Encoding != nil {
```

Comment:
The same comment about stage where custom base64 encoder function set. We can set p.rawUrlBase64Encoding to  base64.RawURLEncoding or custom function on Parser creation

Reply:
Yes, that sounds reasonable.